src/Public/FavIcon.ps1

```diff
@@ -0,0 +1,346 @@
+                Bytes       = [System.IO.File]::ReadAllBytes([System.IO.Path]::Combine($podeRoot, 'favicon.ico'))
+                ContentType = 'image/x-icon'
+            }
+            break
+        }
+    }
+
+    # Auto-detect content type of the image from its byte array
+    $FaviconData.ContentType = Get-PodeImageContentType -Image $FaviconData.Bytes
+
+    # Resolve endpoint targets: single specified or all
+    $keys = if ([string]::IsNullOrEmpty($EndpointName)) {
+        $PodeContext.Server.Endpoints.Keys
+    }
+    else {
+        @($EndpointName)
+    }
+
+    foreach ($key in $keys) {
+        # Only attach favicon to Http or Https endpoints
+        $endpoint = $PodeContext.Server.Endpoints[$key]
+        if ($DefaultEndpoint -and !$endpoint.Default) {
+            continue
+        }
+        if (@('Http', 'Https') -icontains $endpoint.Protocol) {
+            $Endpoint.Favicon = $FaviconData
+        }
+    }
+}
+
+<#
+.SYNOPSIS
+	Checks whether a favicon is configured for one or more Pode endpoints.
+
+.DESCRIPTION
+	This function determines if a favicon is configured on a specific HTTP/S endpoint, or across all
+	endpoints if no name is supplied. You can also limit the check to only endpoints marked as default.
+	It returns $true only if all applicable endpoints have a favicon configured.
+
+.PARAMETER EndpointName
+	The name of the specific endpoint to check. If not provided, all endpoints are checked.
+
+.PARAMETER DefaultEndpoint
+	If supplied, only endpoints marked with `.Default = $true` are checked.
+
+.OUTPUTS
+	System.Boolean
+
+.EXAMPLE
+	Test-PodeFavicon
+
+	# Returns true if all endpoints have a favicon configured.
+
+.EXAMPLE
+	Test-PodeFavicon -EndpointName 'api'
+
+	# Returns true if the 'api' endpoint has a favicon configured.
+
+.EXAMPLE
+	Test-PodeFavicon -DefaultEndpoint
+
+	# Returns true only if all default endpoints have a favicon.
+
+.NOTES
+	This is an internal Pode function and is subject to change.
+#>
+function Test-PodeFavicon {
+    param(
+        [string]
+        $EndpointName,
+
+        [switch]
+        $DefaultEndpoint
+    )
+
+    # Validate endpoint name if supplied
+    if (! [string]::IsNullOrEmpty($EndpointName) -and (! $PodeContext.Server.Endpoints.ContainsKey($EndpointName))) {
+        throw ($Podelocale.endpointNameNotExistExceptionMessage -f $EndpointName)
+    }
+
+    # Collect endpoints to evaluate
+    $keys = if ([string]::IsNullOrEmpty($EndpointName)) {
+        $PodeContext.Server.Endpoints.Keys
+    }
+    else {
+        @($EndpointName)
+    }
+
+    foreach ($key in $keys) {
+        $endpoint = $PodeContext.Server.Endpoints[$key]
+
+        # If filtering to default endpoints only, skip others
+        if ($DefaultEndpoint -and !$endpoint.Default) {
+            continue
+        }
+
+        # If any matched endpoint lacks a favicon, return false
+        if ($null -eq $endpoint.Favicon) {
+            return $false
+        }
+    }
+
+    return $true
```

Comment:
Lines 186-200 for the `foreach` and `return` will want tweaking to the following - so we skip non-HTTP endpoints and return false more appropriately when no default/HTTP endpoints are found:

```powershell
$found = $false
foreach ($key in $keys) {
    $endpoint = $PodeContext.Server.Endpoints[$key]

    # Only check HTTP/S endpoints
    if (@('Http', 'Https') -inotcontains $endpoint.Protocol) {
        continue
    }

    # If filtering to default endpoints only, skip others
    if ($DefaultEndpoint -and !$endpoint.Default) {
        continue
    }

    # If the endpoint has no favicon, return false
    if ($null -eq $endpoint.Favicon) {
        return $false
    }

    # If we reach here, the endpoint has a favicon
    $found = $true
}

return $found
```